avalon/blender/pipeline.py

```diff
@@ -0,0 +1,446 @@
+
+    Returns:
+        The container assembly
+
+    """
+
+    node_name = f"{context['asset']['name']}_{name}"
+    if namespace:
+        node_name = f"{namespace}:{node_name}"
+    if suffix:
+        node_name = f"{node_name}_{suffix}"
+    container = bpy.data.collections.new(name=node_name)
+    # Link the children nodes
+    for obj in nodes:
+        container.objects.link(obj)
+
+    data = {
+        "schema": "avalon-core:container-2.0",
+        "id": AVALON_CONTAINER_ID,
+        "name": name,
+        "namespace": namespace or '',
+        "loader": str(loader),
+        "representation": str(context["representation"]["_id"]),
+    }
+
+    metadata_update(container, data)
+    add_to_avalon_container(container)
+
+    return container
+
+
+def containerise_existing(
+        container: bpy.types.Collection,
+        name: str,
+        namespace: str,
+        context: Dict,
+        loader: Optional[str] = None,
+        suffix: Optional[str] = "CON") -> bpy.types.Collection:
+    """Imprint or update container with metadata.
+
+    Arguments:
+        name: Name of resulting assembly
+        namespace: Namespace under which to host container
+        context: Asset information
+        loader: Name of loader used to produce this container.
+        suffix: Suffix of container, defaults to `_CON`.
+
+    Returns:
+        The container assembly
+    """
+
+    node_name = container.name
+    if namespace:
+        node_name = f"{namespace}:{node_name}"
+    if suffix:
+        node_name = f"{node_name}_{suffix}"
+    container.name = node_name
+    data = {
+        "schema": "avalon-core:container-2.0",
+        "id": AVALON_CONTAINER_ID,
+        "name": name,
+        "namespace": namespace or '',
+        "loader": str(loader),
+        "representation": str(context["representation"]["_id"]),
+    }
+
+    metadata_update(container, data)
+    add_to_avalon_container(container)
+
+    return container
+
+
+def parse_container(container: bpy.types.Collection,
+                    validate: bool = True) -> Dict:
+    """Return the container node's full container data.
+
+    Args:
+        container: A container node name.
+        validate: turn the validation for the container on or off
+
+    Returns:
+        The container schema data for this container node.
+
+    """
+
+    data = lib.read(container)
+
+    # Append transient data
+    data["objectName"] = container.name
+
+    if validate:
+        schema.validate(data)
+
+    return data
+
+
+def _ls():
+    return lib.lsattr("id", AVALON_CONTAINER_ID)
+
+
+def ls() -> Iterator:
+    """List containers from active Blender scene.
+
+    This is the host-equivalent of api.ls(), but instead of listing assets on
+    disk, it lists assets already loaded in Blender; once loaded they are
+    called containers.
+    """
+
+    containers = _ls()
+
+    has_metadata_collector = False
+    config = find_host_config(api.registered_config())
```

Comment:
When PR #501 is merged and `find_host_config` is removed this line can be replaced with `find_submodule(api.registered_config(), "blender")` where [`find_submodule` is defined in `colorbleed.lib`](https://github.com/getavalon/core/pull/501/files#diff-95efa4b4467abab66473e907fd11d122R287)